.circleci/config.yml

```diff
@@ -2,13 +2,13 @@ version: 2.1
 
 with_python: &with_python
   docker:
-    - image: circleci/python:3.7
+    - image: cimg/python:3.9
 
 install_poetry: &install_poetry
   run:
     name: Install Poetry
     command: |
-      pip install poetry==1.1.7
+      pip install poetry==1.2.2
```

Comment:
It's strange that sometimes we seem to need to uninstall the lower version of poetry and sometimes it's fine

Reply:
Good point.  I did some more digging and I figured it out.  It was a difference between the images:
circleci/python3.9 - comes with an old poetry version that needs to be uninstalled
cimg/python3.0 - works without uninstalling poetry